examples/ws-api-client.ts

```diff
@@ -0,0 +1,154 @@
+import { DefaultLogger, WebsocketAPIClient } from '../src';
+
+// or
+// import { DefaultLogger, WebsocketClient } from 'bybit-api';
+
+const key = process.env.API_KEY_COM || '1j3PcP9EMjGPbjcjlC';
+const secret =
+  process.env.API_SECRET_COM || 'VlvBfQqMcGKX77XgYcMqZmOR7biwA9ZvaTwk';
+
+async function main() {
+  // Optional
+  const logger = {
+    ...DefaultLogger,
+    // For a more detailed view of the WebsocketClient, enable the `trace` level by uncommenting the below line:
+    // trace: (...params) => console.log('trace', ...params),
+  };
+
+  const wsClient = new WebsocketAPIClient(
+    {
+      key: key,
+      secret: secret,
+      // testnet: true, // Whether to use the testnet environment: https://testnet.bybit.com/app/user/api-management
+      // demoTrading: false, // note: As of Jan 2025, demo trading does NOT support the WS API
+    },
+    logger, // Optional: inject a custom logger
+  );
+
+  // To do: Check if this is how listeners are handled
+
+  wsClient.getWSClient().on('update', (data) => {
+    console.log('raw message received ', JSON.stringify(data));
+  });
+
+  wsClient.getWSClient().on('open', (data) => {
+    console.log('ws connected', data.wsKey);
+  });
+  wsClient.getWSClient().on('reconnect', ({ wsKey }) => {
+    console.log('ws automatically reconnecting.... ', wsKey);
+  });
+  wsClient.getWSClient().on('reconnected', (data) => {
+    console.log('ws has reconnected ', data?.wsKey);
+  });
+  wsClient.getWSClient().on('authenticated', (data) => {
+    console.log('ws has authenticated ', data?.wsKey);
+  });
```

Comment:
Take a look at this:
https://github.com/tiagosiebler/binance/blob/wsapi/examples/WebSockets/ws-api-client.ts#L83-L84

There are some attached by default. I did the same here with Bybit. So I would include a fn like this, but leave it commented out to show it's unnecessary unless you want more logging control:
https://github.com/tiagosiebler/binance/blob/wsapi/examples/WebSockets/ws-api-client.ts#L34-L61